sklearn/neighbors/tests/test_nca.py

```diff
@@ -127,38 +125,39 @@ def test_params_validation():
     rng = np.random.RandomState(42)
 
     # TypeError
-    assert_raises(TypeError, NCA(max_iter='21').fit, X, y)
-    assert_raises(TypeError, NCA(verbose='true').fit, X, y)
-    assert_raises(TypeError, NCA(tol='1').fit, X, y)
-    assert_raises(TypeError, NCA(n_components='invalid').fit, X, y)
-    assert_raises(TypeError, NCA(warm_start=1).fit, X, y)
+    with pytest.raises(TypeError):
+        NCA(max_iter='21').fit(X, y)
+    with pytest.raises(TypeError):
+        NCA(verbose='true').fit(X, y)
+    with pytest.raises(TypeError):
+        NCA(tol='1').fit(X, y)
+    with pytest.raises(TypeError):
+        NCA(n_components='invalid').fit(X, y)
+    with pytest.raises(TypeError):
+        NCA(warm_start=1).fit(X, y)
 
     # ValueError
-    assert_raise_message(ValueError,
-                         "`init` must be 'auto', 'pca', 'lda', 'identity', "
-                         "'random' or a numpy array of shape "
-                         "(n_components, n_features).",
-                         NCA(init=1).fit, X, y)
-    assert_raise_message(ValueError,
-                         '`max_iter`= -1, must be >= 1.',
-                         NCA(max_iter=-1).fit, X, y)
-
+    msg = r"`init` must be 'auto', 'pca', 'lda', 'identity', "
+    msg += r"'random' or a numpy array of shape "
+    msg += r"(n_components, n_features)."
```

Comment:
Instead of string concatenation, we tend to use multi-line strings using the implicit string concat when defining long string variables:


```suggestion
    msg = (
        r"`init` must be 'auto', 'pca', 'lda', 'identity', 'random' or a "
        r"numpy array of shape (n_components, n_features)."
    )
```

This pattern is occurring several times in this PR. Could you please update the others?

Reply:
String concatenations have been changed to multi-line strings

Let me know if there are any further changes required.